### ISSUE TITLE
No motion and no Bell

### ISSUE THREAD

**Issue by mrahmadt** [tsightler/ring-mqtt#881]
### Describe the Bug

Snapshot & live streaming are working fine, Siren also working fine, but no motion sensor nor bell ring sensor are working, iOS push notification is enabled and working in Ring App 

### Steps to Reproduce

Fresh installation, nothing special 

### Expected Behavior

update sensors for motion detection and bell

### Log Output

```shell
Devices:
Doorbell 2 (Battery)
Indoor Cam (2nd Gen) wired

Deleted from Ring App, reauth RingMQTT, restarted HA, nothing changed

Log file attached
```

### Screenshots

![Screenshot 2024-07-30 at 23 54 18](https://github.com/user-attachments/assets/d0a43a8a-9b5c-4c43-bcef-ac29211453eb)
[home-assistant_mqtt_2024-07-30T20-46-36.543Z.log](https://github.com/user-attachments/files/16433446/home-assistant_mqtt_2024-07-30T20-46-36.543Z.log)
![Screenshot 2024-07-31 at 00 01 15](https://github.com/user-attachments/assets/55995a05-8922-4a80-b97c-bf2f7eb54f7c)
[config_entry-mqtt-01J41XBAY05MEQ27XV4D8A7H9G.json](https://github.com/user-attachments/files/16433531/config_entry-mqtt-01J41XBAY05MEQ27XV4D8A7H9G.json)

![Screenshot 2024-07-31 at 00 00 44](https://github.com/user-attachments/assets/d1d5d323-26a7-4ed9-9110-bcb113915e39)

### Config File

```shell
Attached
```

### Install Type

HA Supervisor

### Version

Current version: 5.6.7

### Operating System

Ubuntu 22.04.4 LTS

### Architecture

x86_64

### Machine Details

Physical 86

**Comment by tsightler** [comment]
Firebase push notifications are undergoing a major refactor due to the migration from legacy FCM API to the new FCM v1 APIs.  Tons of work has been done over the last two months to try to get ready for this as it required changes in the registration methods to be implemented in upstream packages (push-receiver), reverse engineering the required registration parameters from the Ring Andriod app, fixing some registration bugs in those upstream packages, and refactoring ring-client-api (another upstream package on which ring-mqtt depends) as well as modifying ring-mqtt to deal with the new push notification format.

All of this work is in the dev branch, so you can try setting the branch option to dev and restarting.  Theoretically it should "just work", but you may need to repeat the "stop ring-mqtt, delete from Ring control center, start ring-mqtt, re-authenticate" steps.

**Comment by mrahmadt** [comment]
Thanks @tsightler for the clarification

**Comment by tsightler** [comment]
Unfortuantely, it appears that, in some cases, Ring is sending the new style push notifications even to the older versions.  It's possible that new accounts, or perhaps account shared with devices that have the lastest Ring app update, are now defaulting to this new format.  The legacy FCP APIs started the decommissioning process last week, so it's not 100% surprising.  Fortunately, the dev work is largely complete, I've been running the dev branch of ring-mqtt which is built with the beta ring-client-api for the last 3 weeks in my production, with zero issues.  Mostly just waiting on the final ring-client-api to leave beta state, which should happen this week, then I can push an updated ring-mqtt.

**Comment by cmos486** [comment]
I have the same problem with a Ring Intercom, since the update a few days ago it does not recognize the calls to the doorbell.

I have tried the fix indicated but at least in the Spanish app or from the Intercom device this option does not appear.

**Comment by donagrejja** [comment]
I have the same issue after updating to latest version (today).
I can see the event in the logs but it doesn't update the motions sensor in HA.

**Comment by jackyaz** [comment]
+1 from me. if i can do anything to help troubleshoot happy to do so

**Comment by tsightler** [comment]
@donagrejja @jackyaz Did you follow the steps in the release notes to reset the push notification registration?

@donagrejja What logs are you referring to as you did not post any logs.  Do you mean the ring-mqtt logs or something else?

**Comment by jackyaz** [comment]
Ah i missed that, let me try now. Thanks!

**Comment by donagrejja** [comment]
Removing the old connected device and then re-registering worked.
Thanks =)

**Comment by tsightler** [comment]
I'm going to go ahead and close this for now.  If anyone has a case where upgrading to version 5.7.0 (or newer) **doesn't** fix the issue, the we can revisit but we have spent tons of effort over the last months getting these new APIs ironed out and so far, this has always fixed the issue and, hopefully, it will not re-occur now that we are using the modern FCM APIs for device registration.

**Comment by jackyaz** [comment]
I had to remove the device and re-auth twice but all seems OK now. Great work!

**Comment by jackyaz** [comment]
It was working but it's stopped again. Do I try removing and re-adding again?

Edit: this fixed it but I will see if it stops again in a couple of days

**Comment by tsightler** [comment]
@jackyaz The first thing I would always do is just restart the addon, having to remove the device from control center and re-register should be a rare occurrence.  It only happens because Ring does not allow updating the FCM token for a device once it has been initially set, or at least only allows it after a period of time.  However, if for any reason, the GCM registration, which is saved as part of updated refresh tokens, stops working, the only way get push notifications to the new FCM token is to remove the device from the Ring side.

Moving to the new FCM APIs required ring-mqtt to acquire a new FCM token, so if Ring refuses the update, the only fix is to remove the old device and re-register.  However, once this happens, it shouldn't be an issue again.

One thing to note, you haven't shared any details of your install type, but if you happened to be on Home Assistant OS 13, which was released about a week ago, it has a major network issue when HA running on a VM, most notably when running on Proxmox or likely any other KVM/QEMU based virtualazation.  This can all kinds of issues, including loss of push notifications.  I had to upgrade to HA OS 13.1 development version, which should be released to stable channel in the coming days, to address this issue for my prod instance.

**Comment by jackyaz** [comment]
I run on Home Assistant Supervised on Debian. I'll try restarting the addon if the issue happens again and report back

**Comment by JHarding86** [comment]
I am running into this issue.

I only first started using this add-on about a week ago. It worked great for about a day, then stopped.

I have followed all of the trouble shooting steps including removing all of the devices from the authorized client list in the Ring Control Center and ultimately uninstalling the add-on completely and I still am not being notified of motion or rings.

I am running Home Assistant OS 13.1 on Proxmox. I just updated to 13.1 today because of the comment above by @tsightler.
Not sure what else to try...

**Comment by adorobis** [comment]
I also have this problem. Came back from holidays and turned out that I'm not getting dings anymore into HA. Have latest version (in Docker) and re-authorized as described in the documentation. Still not getting the dings. Nothing appears in the container logs neither (have them in the Debug level).
Not sure what else to try...
Additional observations: the attribute "Last ding time" gets updated only after the container is restarted. Later it remains on the value that was valid during the restart, despite the ding button being pressed. Looks like the same happens to the snapshots - they get updated only during ring-mqtt container restart.

**Comment by JHarding86** [comment]
@tsightler should I open a new ticket for this? I just tried the troubleshooting steps again, and I am still not seeing the ring or motion updates? I can add a whole new set of logs and information if need be.

**Comment by tsightler** [comment]
@JHarding86 I don't generally respond to closed issues.  I would not recommend opening a 'bug' issue unless you have proven that it is a bug, but you are welcome to discuss in the support forum.  However, it is unlikely that I will be able to help with this as the issue is almost 100% certain to be environmental and all possible troubleshooting/causes for this issue are already documented thus you'll likely need to troubleshoot the issue yourself since there is zero chance I can reproduce.
